internal/connector/internal/handlers/connector_admin.go

```diff
@@ -400,6 +404,114 @@ func (h *ConnectorAdminHandler) GetConnector(writer http.ResponseWriter, request
 	handlers.HandleGet(writer, request, &cfg)
 }
 
+func (h *ConnectorAdminHandler) GetOperation(resource public.Connector, patch private.ConnectorRequest) (phase.ConnectorOperation, *errors.ServiceError) {
```

Comment:
The admin API is basically doing what the user API is doing here. Instead of duplicating the code move it to common file/functions and reuse them from both API handlers.